### DISCUSSION TITLE
Service Worker bundle size in v3

### DISCUSSION THREAD

**Discussion by wKovacs64** [ShafSpecs/remix-pwa#127] [Q&A]
I just noticed the built output of the SW in v3 is significantly larger than v2 and was just curious if this was expected or if something was potentially sneaking into the build that may not belong. The only features I'm using are SW and manifest.

|    | Full | Minified |
|----|------|----------|
| v2 | 8k   | 4k       |
| v3 | 229k | 97k      |

Any thoughts?

**Comment by ShafSpecs** [comment]
Hmm, would consider sizing soon enough. But it is expected, normally, the old service workers were just event listeners and a few functions. Nothing much, maxixmum of a few hundred lines and you were done. Right now, there's a lot more being added. So for example, worker route apis, runtime utilities and listeners.

**Reply by ShafSpecs** [reply]
This could span up to thousands of lines easily
